drivers/flash/flash_rpi_pico.c

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright (c) 2020 Raspberry Pi (Trading) Ltd.
+ * Copyright (c) 2022 Yonatan Schachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stdint.h>
+#include <stddef.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/irq.h>
+
+#include <hardware/flash.h>
+#include <hardware/regs/io_qspi.h>
+#include <hardware/regs/pads_qspi.h>
+#include <hardware/structs/ssi.h>
+#include <hardware/structs/xip_ctrl.h>
+#include <hardware/resets.h>
+#include <pico/bootrom.h>
+
+LOG_MODULE_REGISTER(flash_rpi_pico, CONFIG_FLASH_LOG_LEVEL);
+
+#define DT_DRV_COMPAT raspberrypi_pico_flash_controller
+
+#define PAGE_SIZE 256
+#define SECTOR_SIZE DT_PROP(DT_CHILD(DT_NODELABEL(flash_controller), flash_0), erase_block_size)
```

Comment:
I think those should be named: `PAGE_WRITE_SIZE` and `SECTOR_ERASE_SIZE`.
The problem I have with just `PAGE_SIZE` and `SECTOR_SIZE` is that there is a real naming mess in non volatile memory storage subsystem.
Sometimes page means write block, sometimes erase size. In other instances sector is uses as write/erase or as smaller erase unit, where page consists of sectors.
The `PAGE_WRITE_SIZE` and `SECTOR_ERASE_SIZE` ain't perfect but at least it tells what the thing is used for, so you get less confused when jumping between drivers.